setup.py

```diff
@@ -19,45 +19,45 @@
     url="https://github.com/vortechsa/python-sdk",
     license="Apache License 2.0",
     classifiers=[
-        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.11.0",
         "License :: OSI Approved :: Apache Software License",
         "Operating System :: OS Independent",
     ],
-    python_requires=">=3.7",
+    python_requires=">=3.11",
     install_requires=[
         "flatten-dict==0.4.2",
-        "pandas>=0.25.2",
+        "pandas==2.2.0",
```

Comment:
This feels like a big jump. Should we also release an intermediate SDK version requiring pandas 1.*.* and possibly a python version in between? my concern is that we may end up forcing the SDK consumers either to live with a very old version of things either move everything up to something very new. 

Having said that this is only relevant if customers use the SDK in production code. If it's only ever used in notebooks then it's probably okay to bump everything up

Reply:
There are quite a few big jumps in this PR. I personally don't think it's a problem. Generally the API of packages doesn't change _that_ much over time, and I only needed to make a handful of changes in this codebase to fix any Pandas issues caused by the upgrade.

I personally think we don't need to introduce intermediary versions. Our clients can continue to use an older version of the SDK if they wish. Happy to open this Q up to a wider audience though.

Reply:
I guess that if it is a big problem they will raise it anyway :-)

Reply:
I would really recommend not going back to a situation where pydantic force a specific version of `pydantic` but also `pandas` (previouslt it was fixed but was relaxed when we reached out). In general as a library it is not recommend to create hard-pinned versions since it can it imposible to fit into an enviorment with packages that have other requirements. Only applications should hard-pin. See https://stackoverflow.com/a/44938662 for an example.

Reply:
Hey @thomasfrederikhoeck, thanks for reaching out. This PR was an attempt to fix some outdated packages and bring the developer experience for the SDK up to scratch. Due to the number of changes and the potential risk to users of the SDK we decided not to merge it yet. I've noted your points about pinning specific versions. I agree this is not ideal. 
I'll be revisiting this PR soon to try to tidy it up and address the point you've raised. Cheers